packages/api/src/router/user.ts

```diff
@@ -130,6 +134,125 @@ export const userRouter = createTRPCRouter({
       }
     }),
 
+  payCoinsByUserId: publicProcedure
+    .input(
+      z.object({
+        receiver: z.object({
+          id: z.string(),
+          username: z.string(),
+          avatar: z.nullable(z.string()),
+          discriminator: z.string(),
+        }),
+        sender: z.object({
+          id: z.string(),
+          username: z.string(),
+          avatar: z.nullable(z.string()),
+          discriminator: z.string(),
+        }),
+        coins: z.number(),
+      }),
+    )
+    .mutation(async ({ ctx, input }) => {
+      try {
+        /**
+         * This function set a temp thumbnail for the user
+         */
+        const tempThumbnail = setTempThumbnail(input.receiver);
+
+        /**
+         * Check in the user, have a balance for the transaction
+         */
+        const sender = await ctx.prisma.user.findUnique({
+          where: { discordId: input.sender.id },
+          select: {
+            coins: true,
+          },
+        });
+
+        /**
+         * If sender not exist, create a new user and return error tokens message
+         */
+        if (!sender) {
+          const sender = await ctx.prisma.user.create({
+            name: input.sender.username,
+            discordId: input.sender.id,
+            discordUserName: input.sender.username,
+            discordDiscriminator: input.sender.discriminator,
+            thumbnail: setTempThumbnail(input.sender),
+            coins: 0,
+          });
+
+          if (sender) {
+            return {
+              status: 'error',
+              message: 'You have no Indie Tokens for this transaction',
+            };
+          } else {
+            return {
+              status: 'error',
+              message: '500 Server error',
+            };
+          }
+        }
+
+        /**
+         * If sender, don't have coins, not aprove the transaction else,
+         * transfers coins to reciver and decrement amount in wallet sender
+         */
```

Comment:
```suggestion
        /**
         * If sender doesn't have coins, not aprove the transaction. 
         * Otherwise, transfer the coins to the receiver and to decrement coins from  the sender's wallet
         */
```